package/kernel/lantiq/ltq-temp/src/ltq-temp.c

```diff
@@ -0,0 +1,141 @@
+/* Lantiq xrx200 CPU Temperatur sensor driver for linux
+ *
+ * Copyright (C) 2016 Florian Eckert <Eckert.Florian@googlemail.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version
+ *
+ * This program is distributed in the hope that it will be useful
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, see <http://www.gnu.org/licenses/>
+ */
+
+#include <linux/module.h>
+#include <linux/init.h>
+#include <linux/of_device.h>
+#include <linux/hwmon.h>
+#include <linux/hwmon-sysfs.h>
+
+#include <lantiq_soc.h>
+
+static int ltq_temp_get(int *p_temp)
+{
+	int temp_value;
+	if ( p_temp == NULL )
+		return -EINVAL;
+
+	ltq_temp_read(&temp_value);
+	*p_temp = (temp_value >> 1);
+
+	return 0;
+}
+
+static ssize_t show_temp(struct device *dev,
+                         struct device_attribute *attr, char *buf)
+{
+	int temp;
+
+	if(ltq_temp_get(&temp))
+		return -EINVAL;
+
+	/* scale temp to millidegree */
+	temp = temp * 1000;
+
+	return sprintf(buf, "%d\n", temp);
+}
+
+static DEVICE_ATTR(temp1_input, S_IRUGO, show_temp, NULL);
+
+static struct attribute *ltq_temp_attrs[] = {
+	&dev_attr_temp1_input.attr,
+	NULL
+};
+
+ATTRIBUTE_GROUPS(ltq_temp);
+
+static int ltq_temp_probe(struct platform_device *pdev)
+{
+	int temp_value = 0;
+	int ret;
+	struct device *hwmon_dev;
+
+	hwmon_dev = devm_hwmon_device_register_with_groups(&pdev->dev,
+                                                           "CPU0",
+                                                           NULL,
+                                                           ltq_temp_groups);
+
+	if (IS_ERR(hwmon_dev)) {
+		dev_err(&pdev->dev, "Failed to register as hwmon device");
+		ret = PTR_ERR(hwmon_dev);
+		goto error_hwmon;
+	}
+
+	ltq_temp_enable();
+
+	if(ltq_temp_get(&temp_value)){
+		dev_err(&pdev->dev, "Unable to read temperature");
+		ret= -ENODEV;
+		goto error;
+	}
+
+	dev_info(&pdev->dev, "Current CPU die temperature: %d °C",
+            temp_value);
```

Comment:
@blogic 
I put the functions as you suggest into the ltq-cputemp.c source. Now i get the following error
> Building modules, stage 2.
>  MODPOST 61 modules
> ERROR: "ltq_cgu_membase" [drivers/hwmon/ltq-cputemp.ko] undefined!
> scripts/Makefile.modpost:91:
> recipe for target '__modpost' failed

I think that the ltq_cgu_membase could not be accessed from an kernel modules scope. If the ltq-temp is loaded as an module. 
The functions must be exported with EXPORT_SYMBOL as in the previous driver version to make it loadable as an module
The registers for the temperature sensor are located in the cgu section (it is made but it is so) What should i do?

regards
Flo